### PR TITLE
chore: Avoid submodules for building igraph

### PR DESCRIPTION
**Pull request by krlmlr**
We keep using the submodules, but only as a source of the C files. The process of updating the C core is documented in `src/README.md` .

### CODE REVIEW

**Comment by szhorvat** [comment]
Makes sense to me, though @ntamas is much more qualified to comment on this type of thing :-)

**Review by szhorvat — APPROVED** [review]

File: src/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# igraph/src
+
+## Updating the C sources
+
+1. Check out and update the `cigraph` submodule to the desired version.
+    ```sh
+    git submodule init
+    git submodule update
+    # git submodule magic
```
Comment (szhorvat) [inline src/README.md:9]: Was this line meant to be left in?

Comment (krlmlr) [reply src/README.md:9]: We could replace it by an example to update to a specific SHA. I'm not familiar with `git submodule` syntax.

Comment (szhorvat) [reply src/README.md:9]: Ah, that's what you meant. I'll complement this README file later to make that clear, but feel free to merge anytime.

**Review by ntamas — APPROVED** [review]

**Pull request merged**